app/models/compute_resource.rb

```diff
@@ -77,6 +77,10 @@ def self.providers_requiring_url
     _("Libvirt, oVirt, OpenStack and Rackspace")
   end
 
+  def self.providers_url_mandatory
+    %w(libvirt ovirt openstack rackspace)
```

Comment:
@lzap I originally introduced `providers_url_mandatory` to be able to override it in other plugins(example: https://github.com/theforeman/foreman-docker/pull/190)
I had thought of doing it the way @timogoebel suggested https://github.com/theforeman/foreman/pull/4461#discussion_r112439805.

But right now I am trying to find an answer for @domcleal's question: http://projects.theforeman.org/issues/19326#note-2
Do you guys see any reason for making `url` as required attribute for VMware?

---
I prefer to keep url validation per model, will do the change.